src/vs/workbench/parts/extensions/electron-browser/extensionEditor.ts

```diff
@@ -292,23 +305,33 @@ export class ExtensionEditor extends BaseEditor {
 		let recommendationsData = {};
 		if (extRecommendations[extension.id.toLowerCase()]) {
 			addClass(this.header, 'recommended');
-			this.recommendation.textContent = extRecommendations[extension.id.toLowerCase()].reasonText;
+			this.recommendationText.textContent = extRecommendations[extension.id.toLowerCase()].reasonText;
 			recommendationsData = { recommendationReason: extRecommendations[extension.id.toLowerCase()].reasonId };
 		} else {
 			removeClass(this.header, 'recommended');
-			this.recommendation.textContent = '';
+			this.recommendationText.textContent = '';
 		}
 
 		/* __GDPR__
-			"extensionGallery:openExtension" : {
-				"recommendationReason": { "classification": "SystemMetaData", "purpose": "FeatureInsight", "isMeasurement": true },
-				"${include}": [
-					"${GalleryExtensionTelemetryData}"
-				]
-			}
+		"extensionGallery:openExtension" : {
+			"recommendationReason": { "classification": "SystemMetaData", "purpose": "FeatureInsight", "isMeasurement": true },
+			"${include}": [
+				"${GalleryExtensionTelemetryData}"
+			]
+		}
 		*/
 		this.telemetryService.publicLog('extensionGallery:openExtension', assign(extension.telemetryData, recommendationsData));
 
+		const ignoredRecommendations = this.extensionTipsService.getAllIgnoredRecommendations();
+
+		toggleClass(this.header, 'ignored', this.recentlyIgnored.indexOf(extension.id.toLowerCase()) !== -1 || ignoredRecommendations.workspace.indexOf(extension.id.toLowerCase()) !== -1);
+		if (this.recentlyIgnored.indexOf(extension.id.toLowerCase()) !== -1) {
```

Comment:
I believe we dont need to maintain the recentlyIgnored list.
As soon as a recommendation is ignored, we update the text in the callback. That should be enough.